DependencyInjection/Compiler/FormFactoryCompilerPass.php

```diff
@@ -40,17 +41,19 @@ public function process(ContainerBuilder $container)
         }
 
         // get factories
-        $original = $container->getDefinition('form.extension');
-        parent::process($container);
+        if (version_compare(Kernel::VERSION, '2.8', '<')) {
```

Comment:
Please read this: https://symfony.com/doc/current/setup/bundles.html#updating-your-code-to-support-symfony-2-x-and-3-x-at-the-same-time

Reply:
I used `version_compare Kernel::VERSION` because:
1) it's the first approach inside CompilerPass that I've seen in other CoreBundle files. 
2) this service is mainly used in the `SonataListFormMappingCommand` that through `isEnabled` checks the symfony version and not the presence of methods...

I think that better rewrite can be made further considering BC behavior from a global perspective...

Reply:
> it's the first approach inside CompilerPass that I've seen in other CoreBundle files.

Mistakes have been made I'm afraid...

Reply:
No problem dude :-) this is my first commit and I'm worrying about standard etc. that I've already messed.

thanks for your patience!

Reply:
:) please read my last comment though, I think it's the most important: https://github.com/sonata-project/SonataCoreBundle/pull/456#issuecomment-341375779

Reply:
BTW, we no longer support symfony versions < 2.8 🤔